docs/guides/manage-application-access-to-streams.md

```diff
@@ -0,0 +1,28 @@
+---
+page_title: "Managing application access to streams"
+---
+
+Managing access through Terraform is not straight forward.
+
+Important concepts to note first,
+- application_access_grant: resource to make request for application to access(consume/produce) a stream in an environment.
+- application_access_grant_approval: resource to approve and revoke application_access_grant
+- application_access_grant_rejection: resource to reject application_access_grant
+
+
+#### Application Owner
+An Application owner can request access to a Strem by creating an `application_access_grant` resource.
+
+- If the `application_access_grant` is auto-approved in the specified environment, then no further action is required on the Application Owner's part. Access is fully granted.
+- If the `application_access_grant` requires approval from Stream Owner, the section on Stream Owner shows them how to do it.
+
+An `application_access_grant` can be canceled by deleting the resource. This is only possible if it is not pending approval.
+
+#### Stream Owner
+A Stream owner can reject access to a Strem by creating a `application_access_grant_rejection` resource. An optional reason can be provided.
+
+A Stream owner can approve access to a Strem by creating a `application_access_grant_approval` resource. 
+
+A Stream owner can revoke access to a Stream by deleting the corresponding `application_access_grant_approval` resource.
+- If the stream request was auto-approved, an `application_access_grant_approval` resource has to be created for the `application_access_grant`, and then destroyed the `application_access_grant_approval` resource.
```

Comment:
Please check your grammar with Grammarly

Reply:
No. You can reject the approval without doing deleting the grant.

Reply:
Why can't I delete `application_access_grant` after deleting the  `application_access_grant_approval `

Reply:
The first scenario is revoking
The second scenario is revoking, then deleting the grant.

Reply:
Making the new request, I have done that according to your suggestion

Reply:
Ok thanks